internal/output/html.go

```diff
@@ -0,0 +1,505 @@
+				Name: packageName,
+			}
+			packageResults[packageName] = packageResult
+		}
+
+		// Get the max severity from groupInfo and increase the count
+		vuln.Summary.SeverityScore = groupInfo.MaxSeverity
+		vuln.Summary.SeverityRating, _ = severity.CalculateRating(vuln.Summary.SeverityScore)
+
+		if _, isUncalled := uncalledVulnIDs[vuln.Summary.ID]; isUncalled {
+			packageResult.UncalledVulns = append(packageResult.UncalledVulns, vuln)
+			packageResult.HTMLVulnCount.Uncalled = len(packageResult.UncalledVulns)
+
+			continue
+		}
+
+		packageResult.CalledVulns = append(packageResult.CalledVulns, vuln)
+		packageResult.HTMLVulnCount.Called = len(packageResult.CalledVulns)
+		addCount(&packageResult.HTMLVulnCount, vuln.Summary.SeverityRating)
+		if vuln.Summary.FixedVersion == UnfixedDescription {
+			packageResult.HTMLVulnCount.UnFixed += 1
+		} else {
+			packageResult.HTMLVulnCount.Fixed += 1
+		}
+	}
+
+	results := make([]HTMLPackageResult, 0, len(packageResults))
+	for _, result := range packageResults {
+		if len(result.CalledVulns) > 0 {
+			result.InstalledVersion = result.CalledVulns[0].Summary.InstalledVersion
+			result.FixedVersion = getMaxFixedVersion(ecosystemPrefix, result.CalledVulns)
+		}
+
+		results = append(results, *result)
+	}
+
+	// Sort packageResults to ensure consistent output
+	sort.Slice(results, func(i, j int) bool {
+		return results[i].Name < results[j].Name
+	})
+
+	return results
+}
+
+// processVulnerabilities processes vulnerabilities for a package
+// and returns a slice of HTMLVulnResult.
+func processVulnerabilities(vulnPkg models.PackageVulns) []HTMLVulnResult {
+	vulnResults := make([]HTMLVulnResult, len(vulnPkg.Vulnerabilities))
+	for i, vuln := range vulnPkg.Vulnerabilities {
+		aliases := vuln.Aliases
+		vulnDetails := HTMLVulnResultDetail{
+			Aliases:     aliases,
+			Description: vuln.Details,
+		}
+		if vulnPkg.Package.ImageOrigin != nil {
+			vulnDetails.LayerCommand, vulnDetails.LayerCommandTooltip = formatLayerCommand(vulnPkg.Package.ImageOrigin.OriginCommand)
+			vulnDetails.LayerID = vulnPkg.Package.ImageOrigin.LayerID
+			vulnDetails.InBaseImage = strconv.FormatBool(vulnPkg.Package.ImageOrigin.InBaseImage)
+		}
+
+		fixedVersion := getFixVersion(vuln.Affected, vulnPkg.Package.Version, vulnPkg.Package.Name, models.Ecosystem(vulnPkg.Package.Ecosystem))
+
+		vulnResults[i] = HTMLVulnResult{
+			Summary: HTMLVulnResultSummary{
+				ID:               vuln.ID,
+				PackageName:      vulnPkg.Package.Name,
+				InstalledVersion: vulnPkg.Package.Version,
+				FixedVersion:     fixedVersion,
+			},
+			Detail: vulnDetails,
+		}
+	}
+
+	return vulnResults
+}
+
+// processVulnerabilityGroups processes vulnerability groups and IDs,
+// populating the called and uncalled maps.
+func processVulnerabilityGroups(groupIDs map[string]models.GroupInfo, vulnPkg models.PackageVulns, uncalledVulnIDs map[string]bool, calledPackages map[string]bool, uncalledPackages map[string]bool) {
```

Comment:
nit: groupIDs should go after vulnPkg. Logic is vulnPkg contains the information you are trying to distribute to the arguments after it.